polyfill/lib/ecmascript.mjs

```diff
@@ -2431,6 +2431,72 @@ export const ES = ObjectAssign({}, ES2020, {
     ES.RejectInstantRange(result);
     return result;
   },
+  AddZonedDateTime: (instant, timeZone, calendar, years, months, weeks, days, h, min, s, ms, µs, ns, overflow) => {
+    // If only time is to be added, then use Instant math. It's not OK to fall
+    // through to the date/time code below because compatible disambiguation in
+    // the PlainDateTime=>Instant conversion will change the offset of any
+    // ZonedDateTtime in the repeated clock time after a backwards transition.
+    // When adding/subtracting time units and not dates, this disambiguation is
+    // not expected and so is avoided below via a fast path for time-only
+    // arithmetic.
+    // BTW, this behavior is similar in spirit to offset: 'prefer' in `with`.
+    if (ES.DurationSign(years, months, weeks, days, 0, 0, 0, 0, 0, 0) === 0) {
+      return ES.AddInstant(GetSlot(instant, EPOCHNANOSECONDS), h, min, s, ms, µs, ns);
+    }
+
+    // RFC 5545 requires the date portion to be added in calendar days and the
+    // time portion to be added in exact time.
+    // FIXME: "op" and the dateAdd/dateSubtract conditional will not be needed
```

Comment:
This comment doesn't match the code below. What conditional?

Reply:
Sorry, left over from a previous version where I was trying to keep both AddZonedDateTime and SubtractZonedDateTime consolidated into one function. I'll remove this.

Reply:
Cool. Please leave the note about #993 though!  Will be helpful for me to find places to fix along with #993. I'm assuming that these AOs should be consolidated the #993 fix-- LMK if not.

Reply:
BTW, I'm hoping to finish the #993 fix today. Tomorrow at latest.

Reply:
There is still a note about #993 in SubtractZonedDateTime, so that should be fine.